lib/rspec/mocks/error_generator.rb

```diff
@@ -288,11 +287,12 @@ def arg_has_valid_description?(arg)
       end
 
       def format_received_args(*args)
-        args.empty? ? "(no args)" : "(" + received_arg_list(*args) + ")"
-      end
-
-      def received_arg_list(*args)
-        args.map(&:inspect).join(", ")
+        return "(no args)" if args.empty? || args.first.empty?
+        return formatted_single_message(args) if single_message?(args)
+        grouped_list = grouped_args(args).map do |arg, index|
+          [formatted_single_message([arg]), group_count(index, args)].join
+        end
+        grouped_list.join("\n            ")
       end
```

Comment:
Utilizing `format_args` makes your example work perfectly.  Similar to my comment above, what do you think of using `args_for_calls` and `args_for_call` vs `args_for_multiple_calls` and `args_for_one_call`.  This might be used on a collection with just one call, so something like `args_for_calls` could refer to single or multiple calls.  Thoughts?